rootfs/fluentd/sbin/stores/stores

```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 source /fluentd/sbin/stores/deis
-source /fluentd/sbin/stores/elastic_search
+[ -e "${ELASTICSEARCH_DEIS_ROUTER_LOGS_ONLY}" ] && source /fluentd/sbin/stores/elastic_search
```

Comment:
What does this line change actually do? When I run this line, it gives me an error status code. Are you sure this actually does what it is supposed to?

Reply:
I should clarify... This gives me an error status code no matter if it is set/unset or if it is True or False. It always give me an error status code on my shell...

Reply:
Sorry about this, this was fixed in another branch I was using but didn't merge it to this branch! `-e` evaluates if a file exists. I foolishly mistook it as 'empty'. But actually 'empty' is `-z` (string is zero size).

I have fixed this

Reply:
> What does this line change actually do? When I run this line, it gives me an error status code. Are you sure this actually does what it is supposed to?

@Cryptophobia 

This is used so we  don't store anything else but the router logs in Elasticsearch. Otherwise,  a second `<store> @type elasticsearch </store>` would be set in `<match **>` and would cause that everything (and the router logs) be logged in elasticsearch.

When `ELASTICSEARCH_DEIS_ROUTER_LOGS_ONLY` is set, then a new `<match>` tag is created for the router logs, and they are forwarded to elasticsearch and deis plugins.

Then, when configuring the more broad `<match **>` tag (this file in question), we evaluate if `ELASTICSEARCH_DEIS_ROUTER_LOGS_ONLY` is set. If it is, then we don't  configure the elasticsearch plugin in this tag, because we just want the router logs.

If `ELASTICSEARCH_DEIS_ROUTER_LOGS_ONLY` is not set, and the `ELASTICSEARCH_` variables are configured, then there is no new `<match>` tag for the router logs, and the elasticsearch plugin is configured in the more broad and default `<match **>` tag. This is the old behavior.

Here is a vague pseudo-config on how the fluentd configuration would look like in each situation

### WITHOUT any `ELASTICSEARCH_*`

```xml
<match **>
  type copy
  <store>
     @type deis
  </store>
</match>
```

### WITH  `ELASTICSEARCH_*` vars set but no `ELASTICSEARCH_DEIS_ROUTER_LOGS_ONLY` (old behavior)

```xml
<match **>
  type copy
  <store>
    @type elasticsearch
  </store>
  <store>
     @type deis
  </store>
</match>
```

### WITH  `ELASTICSEARCH_*` vars set AND `ELASTICSEARCH_DEIS_ROUTER_LOGS_ONLY` set as well

```xml
<match kubernetes.var.log.containers.deis-router-**.log>
  type copy
  <store>
    @type elasticsearch 
  </store>
  <store>
    @type deis
  </store>
</match>
<match **> 
  type copy 
  <store>
     @type deis 
  </store>
</match>
```